File: backend/models/UserGame.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const User = require('./User');

const UserGame = sequelize.define('UserGame', {
  rawgGameId: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('Completed', 'Playing', 'Saved'),
    allowNull: false,
  },
}, {
  tableName: 'usergame',
  freezeTableName: true,
  timestamps: false,
});

// Relaciones
User.hasMany(UserGame, { foreignKey: 'userId', onDelete: 'CASCADE' });
UserGame.belongsTo(User, { foreignKey: 'userId' });

module.exports = UserGame;

File: backend/app/models/user_game.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Boolean, UniqueConstraint
from sqlalchemy.orm import relationship
from datetime import datetime
from app.core.database import Base

class UserGame(Base):
    __tablename__ = "user_games"

    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    game_id = Column(Integer, ForeignKey("games.id", ondelete="CASCADE"), nullable=False)

    status = Column(String)
    score = Column(Integer)
    notes = Column(String)
    added_at = Column(DateTime, default=datetime.utcnow)
    review_updated_at = Column(DateTime(timezone=True), nullable=True)
    contains_spoilers = Column(Boolean, nullable=False, default=False, server_default="false")

    __table_args__ = (
        UniqueConstraint("user_id", "game_id", name="uq_user_games_user_game"),
    )

    user = relationship("User", back_populates="games")
    game = relationship("Game")